web/modules/custom/custom_page/src/Plugin/Block/MenuSubPages.php

```diff
@@ -0,0 +1,242 @@
+   *   The plugin_id for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
+   *   The entity repository service.
+   * @param \Drupal\joinup_core\JoinupRelationManager $relation_manager
+   *   The joinup relation manager service.
+   * @param \Drupal\Core\Plugin\Context\ContextProviderInterface $collection_context
+   *   The collection context.
+   * @param \Drupal\Core\Menu\MenuLinkManagerInterface $menu_link_manager
+   *   The menu link tree service.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match service.
+   * @param \Drupal\og\MembershipManagerInterface $membership_manager
+   *   The og membership manager service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, EntityRepositoryInterface $entity_repository, JoinupRelationManager $relation_manager, ContextProviderInterface $collection_context, MenuLinkManagerInterface $menu_link_manager, RouteMatchInterface $route_match, MembershipManagerInterface $membership_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityRepository = $entity_repository;
+    $this->joinupCoreRelationsManager = $relation_manager;
+    $this->menuLinkManager = $menu_link_manager;
+    $this->currentRouteMatch = $route_match;
+    $this->membershipManager = $membership_manager;
+
+    $collection_contexts = $collection_context->getRuntimeContexts(['og']);
+    if ($collection_contexts && $collection_contexts['og']->hasContextValue()) {
+      $this->collection = $collection_contexts['og']->getContextValue();
+      $results = $this->membershipManager->getGroupContentIds($this->collection, ['ogmenu_instance']);
+
+      if (!empty($results)) {
+        $this->ogMenuInstanceId = reset($results['ogmenu_instance']);
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('entity.repository'),
+      $container->get('joinup_core.relations_manager'),
+      $container->get('collection.collection_route_context'),
+      $container->get('plugin.manager.menu.link'),
+      $container->get('current_route_match'),
+      $container->get('og.membership_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $build = [];
+    $items = [];
+    $child_links = $this->getChildLinks();
+    if (empty($child_links)) {
```

Comment:
Can you put here the discussion we had about the usefulness of returning an empty array for caching purposes?